honeybee_radiance/writer.py

```diff
@@ -1,22 +1,177 @@
+    for shd in aperture.shades:
+        ap_strs.append(shade_to_rad(shd, blk, minimal))
+    return '\n\n'.join(ap_strs)
+
+
+def face_to_rad(face, blk=False, minimal=False):
+    """Get Face as a Radiance string.
+
+    Note that the resulting string does not include full modifier definitions
+    but it does include any of the shades assigned to the Face along with the
+    Apertures and Doors in the Face.
+    
+    Args:
+        face: A honeyee Face for which a RAD representation will be returned.
+        blk: Boolean to note whether the "blacked out" version of the Face should
+            be output, which is useful for direct studies and isolation studies
+            to understand the contribution of individual apertures.
+        minimal: Boolean to note whether the radiance string should be written
+            in a minimal format (with spaces instead of line breaks). Default: False.
+    """
     rad_prop = face.properties.radiance
-    if direct:
-        modifier = rad_prop.modifier_dir
-    else:
-        modifier = rad_prop.modifier
-
-    # TODO: Replace with default modifier set
-    modifier_name = modifier.name if modifier else 'unassigned'
-
-    rad_string = '%s polygon %s\n0\n0\n%d %s' % (
-        modifier_name,
-        face.name,
-        len(face.vertices) * 3,
-        ' '.join('\t%f %f %f\n' % (v.x, v.y, v.z) for v in face.vertices)
-    )
-
-    return rad_string
+    modifier = rad_prop.modifier_blk if blk else rad_prop.modifier
+    rad_poly = Polygon(face.name, face.punched_vertices, modifier)
+    face_strs = [rad_poly.to_radiance(minimal, False, False)]
+    for shd in face.shades:
+        face_strs.append(shade_to_rad(shd, blk, minimal))
+    for dr in face.doors:
+        face_strs.append(door_to_rad(dr, blk, minimal))
+    for ap in face.apertures:
+        face_strs.append(aperture_to_rad(ap, blk, minimal))
+    return '\n\n'.join(face_strs)
+
+
+def room_to_rad(room, blk=False, minimal=False):
+    """Generate an RAD string representation of a Room.
+
+    This method will write all geometry associated with a Room including all
+    Faces, Apertures, Doors, and Shades.
+
+    Args:
+        room: A honeyee Room for which an RAD representation will be returned.
+        blk: Boolean to note whether the "blacked out" version of the geometry
+            should be output, which is useful for direct studies and isolation
+            studies to understand the contribution of individual apertures.
+        minimal: Boolean to note whether the radiance string should be written
+            in a minimal format (with spaces instead of line breaks). Default: False.
+    """
+    room_strs = []
+    for face in room.faces:
+        room_strs.append(face_to_rad(face, blk, minimal))
+    for shd in room.shades:
+        room_strs.append(shade_to_rad(shd, blk, minimal))
+    return '\n\n'.join(room_strs)
+
+
+def model_to_rad(model, blk=False, minimal=False):
```

Comment:
Parts of the writer module will probably change after we talk about the folder structure.

Reply:
So we decided in our talk that the current `to.rad` methods can remain the way that they are (dumping the whole object into a single Rad string). However, we will add a separate `Model.to.rad_folder` method to the writer that spits out the model in the folder structure.